tests/virtualization/universal/waitfor_guests.pm

```diff
@@ -34,19 +34,43 @@ sub run {
     # Wait for guests to announce that installation is complete
     my $retry = 35;
     my $count = 0;
-    while ($count++ < $retry) {
-        my @wait_guests = ();
+
+    my @wait_stage_1_install = @guests;
+    my @wait_stage_2_install = @guests;
```

Comment:
Is it correct understanding that for regular tests without sev_es, only @wait_stage_2_install is used and stage_1 is useless? 
BTW, to me, this part logic is not easy to tell, maybe useful to add a comment to explain the logic for easier maintenance in future.